src/fidesapi/routes/scan.py

```diff
@@ -0,0 +1,122 @@
+from fideslang.models import Organization
+from loguru import logger as log
+from pydantic import BaseModel
+
+from fidesapi.routes.crud import get_resource
+from fidesapi.routes.util import API_PREFIX, unobscure_string
+from fidesapi.sql_models import sql_model_map
+from fidesapi.utils import errors
+from fidesctl.core.system import generate_aws_systems
+
+
+class ScanTypeEnum(str, Enum):
+    """
+    Scan Type Enum to capture the discrete possible values
+    for a valid scan type
+    """
+
+    SYSTEM = "system"
+    DATASET = "dataset"
+
+
+class AWSConfig(BaseModel):
+    """
+    The model for the connection config for AWS
+    """
+
+    region_name: str
+    aws_secret_access_key: str
+    aws_access_key_id: str
+
+
+class OktaConfig(BaseModel):
+    """
+    The model for the connection config for Okta
+    """
+
+    okta_client_token: str
+
+
+class Scan(BaseModel):
+    """
+    The model for the request body housing scan information.
+    """
+
+    organization_key: str
+    scan_type: ScanTypeEnum
+    scan_target: str
+    scan_config: Union[AWSConfig, OktaConfig]
+
+
+router = APIRouter(tags=["scan"], prefix=f"{API_PREFIX}/scan")
+
+routers = [router]
+
+
+@router.post("/", status_code=status.HTTP_201_CREATED)
+async def generate_scan(scan_resource: Scan, response: Response) -> Dict:
+    """
+    Kicks off a generate command for fidesctl.
+
+    Starting off, this will only take 'generate system aws'
+
+    Debate is still up for if one endpoint makes sense or many, might be more
+    difficult to document etc. with all of the config options.
+
+    Currently follows the same logic as `generate_system_aws` in `system.py`
+    """
+
+    organization = await get_resource(
+        sql_model_map["organization"], scan_resource.organization_key
+    )
+    if scan_resource.scan_target.lower() == "aws":
+        log.info("Setting config for AWS")
+        output_list_of_dicts = generate_aws(
+            aws_config=cast(AWSConfig, scan_resource.scan_config),
+            organization=organization,
+        )
+
+    response.status_code = (
+        status.HTTP_200_OK
+        if len(output_list_of_dicts) > 0
+        else status.HTTP_204_NO_CONTENT
+    )
+    return {"systems": output_list_of_dicts}
+
+
+def generate_aws(
+    aws_config: AWSConfig, organization: Organization
+) -> List[Dict[str, str]]:
+    """
+    Returns a list of Systems found in AWS.
+    """
+    config = {
+        "region_name": aws_config.region_name,
+        "aws_access_key_id": unobscure_string(aws_config.aws_access_key_id),
+        "aws_secret_access_key": unobscure_string(aws_config.aws_secret_access_key),
+    }
+    try:
+        log.info("Generating systems from AWS")
+        aws_systems = generate_aws_systems(organization=organization, aws_config=config)
+    except:
+        error = errors.FailedConnectionError()
+        log.bind(error=error.detail["error"]).error("Failed to scan AWS")
+        raise error
```

Comment:
This was a bit of a pain point as well so far... using AWS as an example, we currently have it as an optional installation and to get some of the specific error messages we would need to import those optional libraries here as well. It's certainly required to dive deeper to better understand how to provide the best experience but feels like it spools quickly on the face of it

Reply:
I would expect a function called `generate_aws` to require AWS-related dependencies, so I think it's fine to check for those `import`s at the top of the function, and `raise` an exception if they're not present. Then you can `import` whatever you need, write `except` blocks that target more specific exceptions, and `log`/`raise` errors with more useful messaging.

Reply:
Agreed, started going down that path a bit this morning :) might look at creating a follow on issue to improve error handling deeper but will get a decent start on it in here at least 👍🏽  some of this might help as part of #646

Reply:
i.e. - replacing the `echo_red()` commands with error responses and handling the output method in either the api or cli functions

Reply:
Yep. Also seems related to https://github.com/ethyca/fides/issues/490.